### PR TITLE
ci: various improvements

### PR DESCRIPTION
**Pull request by rvolosatovs**
With this PR in place, CI now builds wasm32-wasi and tests it against enarx via `nix`
`nix flake check` now also runs `cargo fmt`, `cargo clippy` and `cargo nextest` (and fails if either fails)

### CODE REVIEW

**Comment by rvolosatovs** [comment]
In case `nix flake check` becomes slow due to this PR, we should remove `nextest` check

File: flake.nix [hunk 1]
```diff
@@ -23,96 +23,107 @@
...
+            }
+            # TODO: Remove once an overlay is created in enarx
+            // (pkgs.lib.optionalAttrs (system != powerpc64le-linux) {
+              "${cargo.toml.package.name}-wasm32-wasi" = wasm32WasiBin;
+            });
+
+          devShells.default = pkgs.mkShell {
+            buildInputs =
+              [
+                pkgs.openssl
```
Comment (rjzak) [inline flake.nix:119]: Why the `pkgs.openssl`?

Comment (rvolosatovs) [reply flake.nix:119]: It's used for test cert generation

Comment (rvolosatovs) [reply flake.nix:119]: https://github.com/profianinc/steward/blob/947ee7f2416ee44e5c61388cd69e593b12cce927/testdata/generate.sh#L2-L10

**Review by rjzak — APPROVED** [review]

**Pull request merged**